dbt/include/teradata/macros/catalog.sql

```diff
@@ -175,13 +234,22 @@
 --        3. else throw exception. Houston we do not have a something we can use.
     
 --        The result of the above is that dbt can, with "laser-precision" fetch metadata for only the relations it needs, rather than the superset of "all the relations in all the schemas in which dbt has relations".
-{% macro teradata__get_catalog_relations_where_clause_sql(relations) -%}
+{% macro teradata__get_catalog_relations_where_clause_sql(relations, view_tmp_tables_mapping) -%}
     where (
         {%- for relation in relations -%}
             {% if relation.schema and relation.identifier %}
                 (
                     upper("table_schema") = upper('{{ relation.schema }}')
-                    and upper("table_name") = upper('{{ relation.identifier }}')
+                    {% if view_tmp_tables_mapping is not none and view_tmp_tables_mapping|length > 0 %}
+                        {% set temp_table = view_tmp_tables_mapping[relation] %}
```

Comment:
If relation is added to view_tmp_tables_mapping means its view and mapping has happened with temp table. In this case, temp_table won't be empty. So checking of temp_table is only safe side but it won't be try at any time. Is this understanding correct?

Reply:
This is required if any unit tests on a model that contains only tables.